src/cov.jl

```diff
@@ -153,3 +153,80 @@ function cor2cov!(C::AbstractMatrix, s::AbstractArray)
     end
     return C
 end
+
+"""
+    CovarianceEstimator
+
+Abstract type for covariance estimators.
+"""
+abstract type CovarianceEstimator end
+
+"""
+    cov(ce::CovarianceEstimator, x::AbstractVector; mean=nothing)
+
+Compute the variance of the vector `x` using estimator `ce`.
+"""
+cov(ce::CovarianceEstimator, x::AbstractVector; mean=nothing) =
+    error("cov is not defined for $(typeof(ce)) and $(typeof(x))")
+
+"""
+    cov(ce::CovarianceEstimator, x::AbstractVector, y::AbstractVector)
+
+Compute the covariance of the vectors `x` and `y` using estimator `ce`.
+"""
+cov(ce::CovarianceEstimator, x::AbstractVector, y::AbstractVector) =
+    error("cov is not defined for $(typeof(ce)), $(typeof(x)) and $(typeof(y))")
+
+"""
+    cov(ce::CovarianceEstimator, X::AbstractMatrix, [w::AbstractWeights]; mean=nothing, dims::Int=1)
+
+Compute the covariance matrix of the matrix `X` along dimension `dims`
+using estimator `ce`. A weighting vector `w` can be specified.
+The keyword `mean` can be
+
+    * `nothing` (default) in which case the mean is estimated then subtracted
+    from the data `X`,
+    * a given number in which case that provided mean is subtracted from the
+    data `X`.
+"""
+cov(ce::CovarianceEstimator, X::AbstractMatrix; dims::Int=1) =
+    error("cov is not defined for $(typeof(ce)) and $(typeof(X))")
+
+cov(ce::CovarianceEstimator, X::AbstractMatrix, w::AbstractWeights; dims::Int=1) =
+    error("cov is not defined for $(typeof(ce)), $(typeof(X)) and $(typeof(w))")
+
+"""
+    SimpleCovariance(corrected::Bool)
+
+Simple covariance estimator. Estimation calls `cov(x; corrected=corrected)`,
+`cov(x, y; corrected=corrected)` or `cov(X, w, dims; corrected=corrected)`
+where `x`, `y` are vectors, `X` is a matrix and `w` is a weighting vector.
+"""
+struct SimpleCovariance <: CovarianceEstimator
+    corrected::Bool
+end
+SimpleCovariance(;corrected::Bool = false) = SimpleCovariance(corrected)
```

Comment:
Maybe make this an inner constructor if `SimpleCovariance(::Bool)` isn't supposed to be used (and adjust docstring).

Reply:
OK, good idea.